[CI] KeywordFieldBlockLoaderTests testBlockLoaderForFieldInObject {preference=Params[syntheticSource=false, preference=DOC_VALUES]} failing

**Build Scans:**
- [elasticsearch-intake #21170 / part1](https://gradle-enterprise.elastic.co/s/wf35mo2myuah2)
- [elasticsearch-pull-request #65853 / part-1](https://gradle-enterprise.elastic.co/s/y6exujsefjq3g)
- [elasticsearch-intake #21168 / part1](https://gradle-enterprise.elastic.co/s/anixdcseiy3qu)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.blockloader.KeywordFieldBlockLoaderTests.testBlockLoaderForFieldInObject {preference=Params[syntheticSource=false, preference=DOC_VALUES]}" -Dtests.seed=EE69191AC8D6B510 -Dtests.locale=en-VC -Dtests.timezone=US/Aleutian -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.mapper.blockloader.KeywordFieldBlockLoaderTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testBlockLoaderForFieldInObject%20%7Bpreference%3DParams%5BsyntheticSource%3Dfalse%2C%20preference%3DDOC_VALUES%5D%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<[[56 51 6c 68 53 44 6e 71 65 54 47 61 48 6a 59 4e 56 52 44 70 47 4b 48 56 68 50 4e 59 64 48 42 6e 69 45 59 74 62 49 64 56 48 58 73 6e 79 6d], [42 4d 45 6b 4c 4b 42 42 51 4f 58 64 6a 7a 41], [67 53 6d 52 4c 6d 79 45 79 50 7a 74 6c 4c 5a 62 61 52 78 65 67 79 52 70 50 56 46 77 75 7a 47 50 64 67 6e 75 4b 5a 76 59 75 41 67 4a 63 77 4b 53 51], [54 72 51 47 78 72 74 53 6e 56 7a 68 6b 53 4a 69 6e 6a 75 44 77 50 50 44 78 48 74 52 67 4a 6e 52 78 6d 76 53 48 6b 69 50 64 51 41 46 59 68 52 6c], [41 41 57 6d 52 66 74 6d 6e 73 49 51 61 76 70 63 79 47 44 66 7a 69 4c 6e 4b 6b 50 64 4a 4a 59 76], [69 76 75 4f 45 77 55 7a 45 4e 65 4c 63 71 67 79 43 46 7a 44 78 74 52 58 63 4a 41], [73 45 6c 50 51 78 79 4f 6b 58 61 7a 69 52 79 63], [6b 5a 52 78 73 42 50 45 50 47 67 59 69 62 41 5a 75 5a 51 55 4c 59 67 42 77 50 43 4e 63 77], [], [56 67 5a 75 72 4f 72 50 52 45 73 77 6f 57 74 52 71 4a 63 42 5a 61 42 6c 76 75 6b], [6b 72 6e 78 64 47 57 56 72 4e 7a 41 45 53 51 72 50 74 4c], [7
[truncated]
```

**Issue Reasons:**
- [main] 3 failures in test testBlockLoaderForFieldInObject {preference=Params[syntheticSource=false, preference=DOC_VALUES]} (18.8% fail rate in 16 executions)
- [main] 2 failures in step part1 (33.3% fail rate in 6 executions)
- [main] 2 failures in pipeline elasticsearch-intake (33.3% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

I encountered a similar error with another test:

```
REPRODUCE WITH:  ./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.blockloader.KeywordFieldBlockLoaderTests.testBlockLoader {preference=Params[syntheticSource=false, preference=DOC_VALUES]}" -Dtests.seed=A0AC0AC988C0B361 -Dtests.locale=ksh -Dtests.timezone=PST8PDT -Druntime.java=24
  2> java.lang.AssertionError: expected:<[[77 5a 49 6d 4e 53 49 46 79 73 53 56 52 44 70 66 50 78 49 4c 6a 59 6f 42 4f 70], [67 46 4a 45 7a 7a 52 48 45 51]]> but was:<[[67 46 4a 45 7a 7a 52 48 45 51], [77 5a 49 6d 4e 53 49 46 79 73 53 56 52 44 70 66 50 78 49 4c 6a 59 6f 42 4f 70]]>
        at __randomizedtesting.SeedInfo.seed([A0AC0AC988C0B361:B0EDA155E2D2C26A]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at org.elasticsearch.index.mapper.BlockLoaderTestCase.runTest(BlockLoaderTestCase.java:171)
        at org.elasticsearch.index.mapper.BlockLoaderTestCase.testBlockLoader(BlockLoaderTestCase.java:119)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:565)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1763)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.base/java.lang.Thread.run(Thread.java:1447)

```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -380,15 +380,9 @@ tests:
 - class: org.elasticsearch.search.CCSDuelIT
   method: testTerminateAfter
   issue: https://github.com/elastic/elasticsearch/issues/126085
-- class: org.elasticsearch.index.mapper.blockloader.KeywordFieldBlockLoaderTests
-  method: testBlockLoaderForFieldInObject {preference=Params[syntheticSource=false, preference=DOC_VALUES]}
-  issue: https://github.com/elastic/elasticsearch/issues/126088
 - class: org.elasticsearch.search.sort.GeoDistanceIT
   method: testDistanceSortingWithUnmappedField
   issue: https://github.com/elastic/elasticsearch/issues/126118
-- class: org.elasticsearch.index.mapper.blockloader.KeywordFieldBlockLoaderTests
-  method: testBlockLoader {preference=Params[syntheticSource=false, preference=DOC_VALUES]}
-  issue: https://github.com/elastic/elasticsearch/issues/126121
 - class: org.elasticsearch.search.basic.SearchWithRandomDisconnectsIT
   method: testSearchWithRandomDisconnects
   issue: https://github.com/elastic/elasticsearch/issues/122707

--- a/server/src/test/java/org/elasticsearch/index/mapper/blockloader/KeywordFieldBlockLoaderTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/blockloader/KeywordFieldBlockLoaderTests.java
@@ -46,7 +46,9 @@ public class KeywordFieldBlockLoaderTests extends BlockLoaderTestCase {
             .filter(Objects::nonNull);
 
         boolean hasDocValues = hasDocValues(fieldMapping, false);
-        boolean useDocValues = params.preference() == MappedFieldType.FieldExtractPreference.NONE || params.syntheticSource();
+        boolean useDocValues = params.preference() == MappedFieldType.FieldExtractPreference.NONE
+            || params.preference() == MappedFieldType.FieldExtractPreference.DOC_VALUES
+            || params.syntheticSource();
         if (hasDocValues && useDocValues) {
             // Sorted and no duplicates
             var resultList = convertValues.andThen(Stream::distinct)

